testing/base/source_gen.cpp

```diff
@@ -161,24 +160,25 @@ auto SourceGen::ClassGenState::GetValidTypeName() -> llvm::StringRef {
 }
 
 // We want to synthesize shuffled type names from the class names that will be
-// declared, but mixing in `i32` and other builtin types. A somewhat hack-y
-// approach, but currently the strategy is to map every type shorter than the
-// `MinClassNameLength` to one of our builtin types and the longer names to
-// class names in the state.
+// declared, but mixing in `i32` and other builtin types.
 //
-// We use non-unique random identifiers to get a sequence of lengths, and map
-// each identifier to one of our desired types as above. This allows repetition
-// as the identifiers repeat. We can also create a larger pool of builtin type
-// spellings eventually and use the different short identifier spellings to
-// select from this pool.
+// We combine a list of fixed types in the `type_use_params` with the list of
+// (un-shuffled) class names that will be defined to form the spelling of all
```

Comment:
The references to shuffling are a bit spread out -- I think it would be clearer to explain the strategy in one place.
```suggestion
// class names that will be defined to form the spelling of all
```